src/pages/exportUsers.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from 'react';
+
+export default function ExportUsersPage() {
+  const [error, setError] = useState<string | null>(null);
+
+  const convertToCSV = (data: any[]) => {
+    if (data.length === 0) return '';
+
+    const headers = Object.keys(data[0]).join(',') + '\n';
+    const rows = data
+      .map((row) =>
+        Object.values(row)
+          .map((value) => `"${value}"`)
+          .join(',')
+      )
+      .join('\n');
+
+    return headers + rows;
+  };
+
+  const handleDownload = async () => {
+    try {
+      setError(null);
+      const response = await fetch('/api/users');
+
+      if (!response.ok) {
+        throw new Error(`Failed to fetch CSV: ${response.statusText}`);
+      }
+
+      const data = await response.json();
+      const csvData = convertToCSV(data);
+      const blob = new Blob([csvData], { type: 'text/csv' });
+      const url = URL.createObjectURL(blob);
+      const link = document.createElement('a');
+      link.href = url;
+      link.download = 'users.csv';
+      document.body.appendChild(link);
+      link.click();
+      document.body.removeChild(link);
+      URL.revokeObjectURL(url);
+    } catch (err) {
+      console.error('Error downloading CSV:', err);
+      setError('Failed to export users. Please try again.');
+    }
+  };
```

Comment:
I don't love the direct DOM manipulation and think you can do without it.
```suggestion

  const handleDownload = async () => {
    try {
      setError(null);
      const response = await fetch('/api/users');

      if (!response.ok) {
        throw new Error(`Failed to fetch CSV: ${response.statusText}`);
      }

      const data = await response.json();
      const csvData = convertToCSV(data);
      const blob = new Blob([csvData], { type: 'text/csv' });
      const url = URL.createObjectURL(blob);
      const link = document.createElement('a');
      link.href = url;
      link.download = 'users.csv';
      link.click();
      URL.revokeObjectURL(url);
    } catch (err) {
      console.error('Error downloading CSV:', err);
      setError('Failed to export users. Please try again.');
    }
  };
```